src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexGroupCollection.cs

```diff
@@ -101,6 +127,115 @@ private Group GetGroupImpl(int groupnum)
             return _groups[groupnum - 1];
         }
 
+        int IList<Group>.IndexOf(Group item)
+        {
+            var comparer = EqualityComparer<Group>.Default;
+            for (int i = 0; i < Count; i++)
+            {
+                if (comparer.Equals(this[i], item))
+                    return i;
+            }
+            return -1;
+        }
+
+        void IList<Group>.Insert(int index, Group item)
+        {
+            throw new NotSupportedException(SR.NotSupported_ReadOnlyCollection);
+        }
+
+        void IList<Group>.RemoveAt(int index)
+        {
+            throw new NotSupportedException(SR.NotSupported_ReadOnlyCollection);
+        }
+
+        Group IList<Group>.this[int index]
+        {
+            get { return this[index]; }
+            set { throw new NotSupportedException(SR.NotSupported_ReadOnlyCollection); }
+        }
+
+        void ICollection<Group>.Add(Group item)
+        {
+            throw new NotSupportedException(SR.NotSupported_ReadOnlyCollection);
+        }
+
+        void ICollection<Group>.Clear()
+        {
+            throw new NotSupportedException(SR.NotSupported_ReadOnlyCollection);
+        }
+
+        bool ICollection<Group>.Contains(Group item)
+        {
+            var comparer = EqualityComparer<Group>.Default;
```

Comment:
Same question about implementing this in terms of IndexOf.  (I'll refrain from commenting on this kind of thing if it crops up again.)